packages/models-library/tests/test_utils_pydantic_tools_extension.py

```diff
@@ -14,15 +14,20 @@ class MyModel(BaseModel):
 
 
 def test_schema():
-    assert MyModel.schema() == {
+    assert MyModel.model_json_schema() == {
```

Comment:
looks like your changes broke the test. Are you sure?

Reply:
Actually this tests is showing that now pydantic NICELY distinguishes nullable fields. NOTE

```python
class MyModel(BaseModel):
    a: int
    b: int | None = Field(...)
    c: int = 42
    d: int | None = None
    e: int = FieldNotRequired(description="optional non-nullable")
```
And we should get an schema that defines:
- `a` is a **required** `int`
- `b` is a **required**  either `int` or `Nullable` 
- `c` is an **optional** `int` that defaults to `42`
- `d` is an **optional**  either `int` or `Nullable`  that defaults to `None`
- `e` is an **optional**  `int` (if not set, it just gives you `None`)